weather_dl/download_pipeline/fetcher.py

```diff
@@ -68,11 +69,18 @@ def fetch_data(self, config: Config, *, worker_name: str = 'default') -> None:
         client = CLIENTS[self.client_name](config)
         target = prepare_target_name(config)
 
-        with self.manifest.transact(config.selection, target, config.user_id):
-            with tempfile.NamedTemporaryFile() as temp:
-                logger.info(f'[{worker_name}] Fetching data for {target!r}.')
+        with tempfile.NamedTemporaryFile() as temp:
+            logger.info(f'[{worker_name}] Fetching data for {target!r}.')
+            with self.manifest.transact(config.kwargs.get('config_name'), config.selection, target, config.user_id):
                 self.retrieve(client, config.dataset, config.selection, temp.name)
 
+                self.manifest.set_stage(Stage.UPLOAD)
+                precise_upload_start_time = (
+                    datetime.datetime.utcnow()
+                    .replace(tzinfo=datetime.timezone.utc)
+                    .isoformat(timespec='seconds')
+                )
+                self.manifest.prev_stage_precise_start_time = precise_upload_start_time
```

Comment:
It seems like this logic gets duplicated a bunch. Is there some way that we can include it in `set_stage`?

Reply:
This was done to reduce the impact of _read() and _update() calls on the stage start time, effectively eliminating the time taken by insert and update query operations.

Reply:
If there's a good technical reason to include this, I'm OK with it.